core/src/consensus/stake/mod.rs

```diff
@@ -95,8 +95,8 @@ impl ActionHandler for Stake {
             } => revoke(state, sender, &address, quantity),
             Action::SelfNominate {
                 deposit,
-                ..
-            } => self_nominate(state, sender, sender_public, deposit, 0, 0),
+                metadata,
+            } => self_nominate(state, sender, sender_public, deposit, 0, 0, &metadata),
```

Comment:
Why do you pass the reference of the metadata and make a new vector in it?
How about moving the vector here?